### PR TITLE
Migration component detection to .net6

### PR DESCRIPTION
**Pull request by RushabhBhansali**
Created based on #76 

Changes from #76 
- upgraded CD build to .net6

Additional work (only in this PR):
- Fixed compilation errors and suppressed the analyzer errors 

### Testing
- successful dotnet build.
- successful unit tests.

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -1,564 +1,271 @@
-# Version: 4.1.1 (Using https://semver.org/)
```
Comment (JamieMagee) [inline .editorconfig:1]: Can you revert your changes to this file and edit it manually? Visual Studio has completely mangled it.

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -25,7 +25,7 @@ jobs:
       - name: Setup .NET
         uses: actions/setup-dotnet@v3
         with:
-          dotnet-version: 3.1.x
```
Comment (JamieMagee) [inline .github/workflows/build.yml:28]: Remove
```yml
with:
  dotnet-version: 6.0.x
```

From all the workflow files changed. The `actions/setup-dotnet` action will read `global.json` to find the SDK version.

File: src/Microsoft.ComponentDetection.Common/ComponentComparer.cs [hunk 3]
```diff
@@ -1,5 +1,5 @@
 ﻿using System.Collections.Generic;
-using Microsoft.ComponentDetection.Contracts.TypedComponent;
+using Microsoft.ComponentDetection.Contracts.TypedComponentNS;
```
Comment (JamieMagee) [inline src/Microsoft.ComponentDetection.Common/ComponentComparer.cs:2]: Revert this namespace change and silence the linter warning if possible.

**Comment by JamieMagee** [comment]
Closes #107

File: .editorconfig (hunk 1 above)
Comment (RushabhBhansali) [reply .editorconfig:1]: yeah, I realize it. going to do that.

Comment (RushabhBhansali) [reply .editorconfig]: done.

File: global.json [hunk 4]
```diff
@@ -1,6 +1,6 @@
 {
 	"sdk": {
-		"version": "3.1.300",
+		"version": "6.0.200",
```
[the remark this answers is not in the document]
Comment (RushabhBhansali) [reply global.json]: fixed

File: .editorconfig [hunk 5]
```diff
@@ -561,4 +561,410 @@ dotnet_diagnostic.CS0618.severity = suggestion
 # Missing XML comment for publicly visible type or member '...'
 dotnet_diagnostic.CS1591.severity = suggestion
 
+# SA1600: Elements should be documented
+dotnet_diagnostic.SA1600.severity = suggestion
```
Comment (JamieMagee) [inline .editorconfig:565]: All of these changes appear to be duplicated above. Can you revert all this and selectively disable only new warnings that were introduced with .NET 6?

File: .github/workflows/build.yml (hunk 2 above)
Comment (RushabhBhansali) [reply .github/workflows/build.yml]: done.

File: src/Microsoft.ComponentDetection.Common/ComponentComparer.cs (hunk 3 above)
Comment (RushabhBhansali) [reply src/Microsoft.ComponentDetection.Common/ComponentComparer.cs]: done. removed all the duplicates in [this commit](https://github.com/microsoft/component-detection/pull/288/commits/fa8fce6651bb6d7fa8e5efa23707563b8b4ecb70)

File: .editorconfig (hunk 5 above)
Comment (RushabhBhansali) [reply .editorconfig]: done. removed all the duplicates in [this commit](https://github.com/microsoft/component-detection/pull/288/commits/fa8fce6651bb6d7fa8e5efa23707563b8b4ecb70)

File: src/Microsoft.ComponentDetection.Common/ComponentComparer.cs (hunk 3 above)
Comment (RushabhBhansali) [reply src/Microsoft.ComponentDetection.Common/ComponentComparer.cs]: done. reverted all the namespace changes and suppressed the warning

**Review by JamieMagee — APPROVED** [review]

**Pull request merged**